presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java

```diff
@@ -266,12 +272,12 @@ public void lockDownNodes()
         @Override
         public List<Node> allNodes()
         {
-            return ImmutableList.copyOf(partitioning.getPartitionToNode().values());
+            return ImmutableList.copyOf(allNodes);
```

Comment:
Put `ImmutableList.copyOf` in constructor

Reply:
@haozhun : Do we want to do defensive copy here as well?

Reply:
No.  If a collection has an immutable defensive copy in constructor, it is returned directly in from getters..